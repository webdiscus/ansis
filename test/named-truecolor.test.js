import { describe, test, expect, vi } from 'vitest';

import colorNames from 'css-color-names';
import ansis, { Ansis } from 'ansis';

// truecolor level
const level = 3;

let getBgName = (name) => 'bg' + name[0].toUpperCase() + name.slice(1);

// mock ansis to ensure it always uses the required colorNames level,
// independent of what would be auto-detected
vi.mock('ansis', async () => {
  const { Ansis } = await vi.importActual('ansis');
  const ansis = new Ansis(3); // force init with level
  return { default: ansis, Ansis }
});

describe('extend with named colors', () => {
  test('foreground pink via hex', async () => {
    // colorNames.pink => #ffc0cb
    const received = ansis.hex(colorNames.pink)('Pink Foreground via HEX');
    const expected = '[38;2;255;192;203mPink Foreground via HEX[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('background pink via hex', async () => {
    // colorNames.pink => #ffc0cb
    const received = ansis.bgHex(colorNames.pink).black('Pink Background via HEX');
    const expected = '[48;2;255;192;203m[30mPink Background via HEX[39m[49m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('foreground helper for named colors', async () => {
    const color = (name) => {
      const hex = colorNames[name.toLowerCase()];
      if (!hex) throw new Error(`unknown color: ${name}`);
      return ansis.hex(hex);
    };

    //const received = color('Orange').underline('Orange Foreground via helper');
    const received = ansis.hex(colorNames['orange']).underline('Orange Foreground via helper');
    const expected = '[38;2;255;165;0m[4mOrange Foreground via helper[24m[39m';

    console.log(received);
    expect(received).toEqual(expected);
  });

  test('background helper for named colors', async () => {
    const bgColor = (name) => {
      const hex = colorNames[name.toLowerCase()];
      if (!hex) throw new Error(`unknown color: ${name}`);
      return ansis.bgHex(hex);
    };

    const received = bgColor('Orange').black.underline('Orange Background via helper');
    const expected = '[48;2;255;165;0m[30m[4mOrange Background via helper[24m[39m[49m';

    console.log(received);
    expect(received).toEqual(expected);
  });

  test('extended via new instance', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const received = ansis.pink('Pink');
    const expected = '[38;2;255;192;203mPink[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('extended via ansis.extend()', async () => {
    const color = ansis.extend(colorNames);
    const received = color.pink('Pink');
    const expected = '[38;2;255;192;203mPink[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('extended bg color', async () => {
    const color = ansis.extend(colorNames);
    const received = color.bgPink.black('Pink Background');
    const expected = '[48;2;255;192;203m[38;2;0;0;0mPink Background[39m[49m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('styled extended colorNames', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const received = ansis.pink.underline('Pink');
    const expected = '[38;2;255;192;203m[4mPink[24m[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('print all named truecolor', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const names = Object.keys(colorNames);
    //const received = names.map((name) => ansis[name](name)).join('\n');
    const received = names.map((name) => {
      const bgName = getBgName(name);
      return ansis[name](`${name} `) + ansis[bgName].black(` ${bgName} `);
    }).join('\n');

    console.log(received);

    expect(received).toMatchSnapshot();
  });
});