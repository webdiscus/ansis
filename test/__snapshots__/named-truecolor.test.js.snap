// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extend with named colors > print all named truecolor 1`] = `
"[38;2;240;248;255maliceblue [39m[48;2;240;248;255m[38;2;0;0;0m bgAliceblue [39m[49m
[38;2;250;235;215mantiquewhite [39m[48;2;250;235;215m[38;2;0;0;0m bgAntiquewhite [39m[49m
[38;2;0;255;255maqua [39m[48;2;0;255;255m[38;2;0;0;0m bgAqua [39m[49m
[38;2;127;255;212maquamarine [39m[48;2;127;255;212m[38;2;0;0;0m bgAquamarine [39m[49m
[38;2;240;255;255mazure [39m[48;2;240;255;255m[38;2;0;0;0m bgAzure [39m[49m
[38;2;245;245;220mbeige [39m[48;2;245;245;220m[38;2;0;0;0m bgBeige [39m[49m
[38;2;255;228;196mbisque [39m[48;2;255;228;196m[38;2;0;0;0m bgBisque [39m[49m
[38;2;0;0;0mblack [39m[48;2;0;0;0m[38;2;0;0;0m bgBlack [39m[49m
[38;2;255;235;205mblanchedalmond [39m[48;2;255;235;205m[38;2;0;0;0m bgBlanchedalmond [39m[49m
[38;2;0;0;255mblue [39m[48;2;0;0;255m[38;2;0;0;0m bgBlue [39m[49m
[38;2;138;43;226mblueviolet [39m[48;2;138;43;226m[38;2;0;0;0m bgBlueviolet [39m[49m
[38;2;165;42;42mbrown [39m[48;2;165;42;42m[38;2;0;0;0m bgBrown [39m[49m
[38;2;222;184;135mburlywood [39m[48;2;222;184;135m[38;2;0;0;0m bgBurlywood [39m[49m
[38;2;95;158;160mcadetblue [39m[48;2;95;158;160m[38;2;0;0;0m bgCadetblue [39m[49m
[38;2;127;255;0mchartreuse [39m[48;2;127;255;0m[38;2;0;0;0m bgChartreuse [39m[49m
[38;2;210;105;30mchocolate [39m[48;2;210;105;30m[38;2;0;0;0m bgChocolate [39m[49m
[38;2;255;127;80mcoral [39m[48;2;255;127;80m[38;2;0;0;0m bgCoral [39m[49m
[38;2;100;149;237mcornflowerblue [39m[48;2;100;149;237m[38;2;0;0;0m bgCornflowerblue [39m[49m
[38;2;255;248;220mcornsilk [39m[48;2;255;248;220m[38;2;0;0;0m bgCornsilk [39m[49m
[38;2;220;20;60mcrimson [39m[48;2;220;20;60m[38;2;0;0;0m bgCrimson [39m[49m
[38;2;0;255;255mcyan [39m[48;2;0;255;255m[38;2;0;0;0m bgCyan [39m[49m
[38;2;0;0;139mdarkblue [39m[48;2;0;0;139m[38;2;0;0;0m bgDarkblue [39m[49m
[38;2;0;139;139mdarkcyan [39m[48;2;0;139;139m[38;2;0;0;0m bgDarkcyan [39m[49m
[38;2;184;134;11mdarkgoldenrod [39m[48;2;184;134;11m[38;2;0;0;0m bgDarkgoldenrod [39m[49m
[38;2;169;169;169mdarkgray [39m[48;2;169;169;169m[38;2;0;0;0m bgDarkgray [39m[49m
[38;2;0;100;0mdarkgreen [39m[48;2;0;100;0m[38;2;0;0;0m bgDarkgreen [39m[49m
[38;2;169;169;169mdarkgrey [39m[48;2;169;169;169m[38;2;0;0;0m bgDarkgrey [39m[49m
[38;2;189;183;107mdarkkhaki [39m[48;2;189;183;107m[38;2;0;0;0m bgDarkkhaki [39m[49m
[38;2;139;0;139mdarkmagenta [39m[48;2;139;0;139m[38;2;0;0;0m bgDarkmagenta [39m[49m
[38;2;85;107;47mdarkolivegreen [39m[48;2;85;107;47m[38;2;0;0;0m bgDarkolivegreen [39m[49m
[38;2;255;140;0mdarkorange [39m[48;2;255;140;0m[38;2;0;0;0m bgDarkorange [39m[49m
[38;2;153;50;204mdarkorchid [39m[48;2;153;50;204m[38;2;0;0;0m bgDarkorchid [39m[49m
[38;2;139;0;0mdarkred [39m[48;2;139;0;0m[38;2;0;0;0m bgDarkred [39m[49m
[38;2;233;150;122mdarksalmon [39m[48;2;233;150;122m[38;2;0;0;0m bgDarksalmon [39m[49m
[38;2;143;188;143mdarkseagreen [39m[48;2;143;188;143m[38;2;0;0;0m bgDarkseagreen [39m[49m
[38;2;72;61;139mdarkslateblue [39m[48;2;72;61;139m[38;2;0;0;0m bgDarkslateblue [39m[49m
[38;2;47;79;79mdarkslategray [39m[48;2;47;79;79m[38;2;0;0;0m bgDarkslategray [39m[49m
[38;2;47;79;79mdarkslategrey [39m[48;2;47;79;79m[38;2;0;0;0m bgDarkslategrey [39m[49m
[38;2;0;206;209mdarkturquoise [39m[48;2;0;206;209m[38;2;0;0;0m bgDarkturquoise [39m[49m
[38;2;148;0;211mdarkviolet [39m[48;2;148;0;211m[38;2;0;0;0m bgDarkviolet [39m[49m
[38;2;255;20;147mdeeppink [39m[48;2;255;20;147m[38;2;0;0;0m bgDeeppink [39m[49m
[38;2;0;191;255mdeepskyblue [39m[48;2;0;191;255m[38;2;0;0;0m bgDeepskyblue [39m[49m
[38;2;105;105;105mdimgray [39m[48;2;105;105;105m[38;2;0;0;0m bgDimgray [39m[49m
[38;2;105;105;105mdimgrey [39m[48;2;105;105;105m[38;2;0;0;0m bgDimgrey [39m[49m
[38;2;30;144;255mdodgerblue [39m[48;2;30;144;255m[38;2;0;0;0m bgDodgerblue [39m[49m
[38;2;178;34;34mfirebrick [39m[48;2;178;34;34m[38;2;0;0;0m bgFirebrick [39m[49m
[38;2;255;250;240mfloralwhite [39m[48;2;255;250;240m[38;2;0;0;0m bgFloralwhite [39m[49m
[38;2;34;139;34mforestgreen [39m[48;2;34;139;34m[38;2;0;0;0m bgForestgreen [39m[49m
[38;2;255;0;255mfuchsia [39m[48;2;255;0;255m[38;2;0;0;0m bgFuchsia [39m[49m
[38;2;220;220;220mgainsboro [39m[48;2;220;220;220m[38;2;0;0;0m bgGainsboro [39m[49m
[38;2;248;248;255mghostwhite [39m[48;2;248;248;255m[38;2;0;0;0m bgGhostwhite [39m[49m
[38;2;218;165;32mgoldenrod [39m[48;2;218;165;32m[38;2;0;0;0m bgGoldenrod [39m[49m
[38;2;255;215;0mgold [39m[48;2;255;215;0m[38;2;0;0;0m bgGold [39m[49m
[38;2;128;128;128mgray [39m[48;2;128;128;128m[38;2;0;0;0m bgGray [39m[49m
[38;2;0;128;0mgreen [39m[48;2;0;128;0m[38;2;0;0;0m bgGreen [39m[49m
[38;2;173;255;47mgreenyellow [39m[48;2;173;255;47m[38;2;0;0;0m bgGreenyellow [39m[49m
[38;2;128;128;128mgrey [39m[48;2;128;128;128m[38;2;0;0;0m bgGrey [39m[49m
[38;2;240;255;240mhoneydew [39m[48;2;240;255;240m[38;2;0;0;0m bgHoneydew [39m[49m
[38;2;255;105;180mhotpink [39m[48;2;255;105;180m[38;2;0;0;0m bgHotpink [39m[49m
[38;2;205;92;92mindianred [39m[48;2;205;92;92m[38;2;0;0;0m bgIndianred [39m[49m
[38;2;75;0;130mindigo [39m[48;2;75;0;130m[38;2;0;0;0m bgIndigo [39m[49m
[38;2;255;255;240mivory [39m[48;2;255;255;240m[38;2;0;0;0m bgIvory [39m[49m
[38;2;240;230;140mkhaki [39m[48;2;240;230;140m[38;2;0;0;0m bgKhaki [39m[49m
[38;2;255;240;245mlavenderblush [39m[48;2;255;240;245m[38;2;0;0;0m bgLavenderblush [39m[49m
[38;2;230;230;250mlavender [39m[48;2;230;230;250m[38;2;0;0;0m bgLavender [39m[49m
[38;2;124;252;0mlawngreen [39m[48;2;124;252;0m[38;2;0;0;0m bgLawngreen [39m[49m
[38;2;255;250;205mlemonchiffon [39m[48;2;255;250;205m[38;2;0;0;0m bgLemonchiffon [39m[49m
[38;2;173;216;230mlightblue [39m[48;2;173;216;230m[38;2;0;0;0m bgLightblue [39m[49m
[38;2;240;128;128mlightcoral [39m[48;2;240;128;128m[38;2;0;0;0m bgLightcoral [39m[49m
[38;2;224;255;255mlightcyan [39m[48;2;224;255;255m[38;2;0;0;0m bgLightcyan [39m[49m
[38;2;250;250;210mlightgoldenrodyellow [39m[48;2;250;250;210m[38;2;0;0;0m bgLightgoldenrodyellow [39m[49m
[38;2;211;211;211mlightgray [39m[48;2;211;211;211m[38;2;0;0;0m bgLightgray [39m[49m
[38;2;144;238;144mlightgreen [39m[48;2;144;238;144m[38;2;0;0;0m bgLightgreen [39m[49m
[38;2;211;211;211mlightgrey [39m[48;2;211;211;211m[38;2;0;0;0m bgLightgrey [39m[49m
[38;2;255;182;193mlightpink [39m[48;2;255;182;193m[38;2;0;0;0m bgLightpink [39m[49m
[38;2;255;160;122mlightsalmon [39m[48;2;255;160;122m[38;2;0;0;0m bgLightsalmon [39m[49m
[38;2;32;178;170mlightseagreen [39m[48;2;32;178;170m[38;2;0;0;0m bgLightseagreen [39m[49m
[38;2;135;206;250mlightskyblue [39m[48;2;135;206;250m[38;2;0;0;0m bgLightskyblue [39m[49m
[38;2;119;136;153mlightslategray [39m[48;2;119;136;153m[38;2;0;0;0m bgLightslategray [39m[49m
[38;2;119;136;153mlightslategrey [39m[48;2;119;136;153m[38;2;0;0;0m bgLightslategrey [39m[49m
[38;2;176;196;222mlightsteelblue [39m[48;2;176;196;222m[38;2;0;0;0m bgLightsteelblue [39m[49m
[38;2;255;255;224mlightyellow [39m[48;2;255;255;224m[38;2;0;0;0m bgLightyellow [39m[49m
[38;2;0;255;0mlime [39m[48;2;0;255;0m[38;2;0;0;0m bgLime [39m[49m
[38;2;50;205;50mlimegreen [39m[48;2;50;205;50m[38;2;0;0;0m bgLimegreen [39m[49m
[38;2;250;240;230mlinen [39m[48;2;250;240;230m[38;2;0;0;0m bgLinen [39m[49m
[38;2;255;0;255mmagenta [39m[48;2;255;0;255m[38;2;0;0;0m bgMagenta [39m[49m
[38;2;128;0;0mmaroon [39m[48;2;128;0;0m[38;2;0;0;0m bgMaroon [39m[49m
[38;2;102;205;170mmediumaquamarine [39m[48;2;102;205;170m[38;2;0;0;0m bgMediumaquamarine [39m[49m
[38;2;0;0;205mmediumblue [39m[48;2;0;0;205m[38;2;0;0;0m bgMediumblue [39m[49m
[38;2;186;85;211mmediumorchid [39m[48;2;186;85;211m[38;2;0;0;0m bgMediumorchid [39m[49m
[38;2;147;112;219mmediumpurple [39m[48;2;147;112;219m[38;2;0;0;0m bgMediumpurple [39m[49m
[38;2;60;179;113mmediumseagreen [39m[48;2;60;179;113m[38;2;0;0;0m bgMediumseagreen [39m[49m
[38;2;123;104;238mmediumslateblue [39m[48;2;123;104;238m[38;2;0;0;0m bgMediumslateblue [39m[49m
[38;2;0;250;154mmediumspringgreen [39m[48;2;0;250;154m[38;2;0;0;0m bgMediumspringgreen [39m[49m
[38;2;72;209;204mmediumturquoise [39m[48;2;72;209;204m[38;2;0;0;0m bgMediumturquoise [39m[49m
[38;2;199;21;133mmediumvioletred [39m[48;2;199;21;133m[38;2;0;0;0m bgMediumvioletred [39m[49m
[38;2;25;25;112mmidnightblue [39m[48;2;25;25;112m[38;2;0;0;0m bgMidnightblue [39m[49m
[38;2;245;255;250mmintcream [39m[48;2;245;255;250m[38;2;0;0;0m bgMintcream [39m[49m
[38;2;255;228;225mmistyrose [39m[48;2;255;228;225m[38;2;0;0;0m bgMistyrose [39m[49m
[38;2;255;228;181mmoccasin [39m[48;2;255;228;181m[38;2;0;0;0m bgMoccasin [39m[49m
[38;2;255;222;173mnavajowhite [39m[48;2;255;222;173m[38;2;0;0;0m bgNavajowhite [39m[49m
[38;2;0;0;128mnavy [39m[48;2;0;0;128m[38;2;0;0;0m bgNavy [39m[49m
[38;2;253;245;230moldlace [39m[48;2;253;245;230m[38;2;0;0;0m bgOldlace [39m[49m
[38;2;128;128;0molive [39m[48;2;128;128;0m[38;2;0;0;0m bgOlive [39m[49m
[38;2;107;142;35molivedrab [39m[48;2;107;142;35m[38;2;0;0;0m bgOlivedrab [39m[49m
[38;2;255;165;0morange [39m[48;2;255;165;0m[38;2;0;0;0m bgOrange [39m[49m
[38;2;255;69;0morangered [39m[48;2;255;69;0m[38;2;0;0;0m bgOrangered [39m[49m
[38;2;218;112;214morchid [39m[48;2;218;112;214m[38;2;0;0;0m bgOrchid [39m[49m
[38;2;238;232;170mpalegoldenrod [39m[48;2;238;232;170m[38;2;0;0;0m bgPalegoldenrod [39m[49m
[38;2;152;251;152mpalegreen [39m[48;2;152;251;152m[38;2;0;0;0m bgPalegreen [39m[49m
[38;2;175;238;238mpaleturquoise [39m[48;2;175;238;238m[38;2;0;0;0m bgPaleturquoise [39m[49m
[38;2;219;112;147mpalevioletred [39m[48;2;219;112;147m[38;2;0;0;0m bgPalevioletred [39m[49m
[38;2;255;239;213mpapayawhip [39m[48;2;255;239;213m[38;2;0;0;0m bgPapayawhip [39m[49m
[38;2;255;218;185mpeachpuff [39m[48;2;255;218;185m[38;2;0;0;0m bgPeachpuff [39m[49m
[38;2;205;133;63mperu [39m[48;2;205;133;63m[38;2;0;0;0m bgPeru [39m[49m
[38;2;255;192;203mpink [39m[48;2;255;192;203m[38;2;0;0;0m bgPink [39m[49m
[38;2;221;160;221mplum [39m[48;2;221;160;221m[38;2;0;0;0m bgPlum [39m[49m
[38;2;176;224;230mpowderblue [39m[48;2;176;224;230m[38;2;0;0;0m bgPowderblue [39m[49m
[38;2;128;0;128mpurple [39m[48;2;128;0;128m[38;2;0;0;0m bgPurple [39m[49m
[38;2;102;51;153mrebeccapurple [39m[48;2;102;51;153m[38;2;0;0;0m bgRebeccapurple [39m[49m
[38;2;255;0;0mred [39m[48;2;255;0;0m[38;2;0;0;0m bgRed [39m[49m
[38;2;188;143;143mrosybrown [39m[48;2;188;143;143m[38;2;0;0;0m bgRosybrown [39m[49m
[38;2;65;105;225mroyalblue [39m[48;2;65;105;225m[38;2;0;0;0m bgRoyalblue [39m[49m
[38;2;139;69;19msaddlebrown [39m[48;2;139;69;19m[38;2;0;0;0m bgSaddlebrown [39m[49m
[38;2;250;128;114msalmon [39m[48;2;250;128;114m[38;2;0;0;0m bgSalmon [39m[49m
[38;2;244;164;96msandybrown [39m[48;2;244;164;96m[38;2;0;0;0m bgSandybrown [39m[49m
[38;2;46;139;87mseagreen [39m[48;2;46;139;87m[38;2;0;0;0m bgSeagreen [39m[49m
[38;2;255;245;238mseashell [39m[48;2;255;245;238m[38;2;0;0;0m bgSeashell [39m[49m
[38;2;160;82;45msienna [39m[48;2;160;82;45m[38;2;0;0;0m bgSienna [39m[49m
[38;2;192;192;192msilver [39m[48;2;192;192;192m[38;2;0;0;0m bgSilver [39m[49m
[38;2;135;206;235mskyblue [39m[48;2;135;206;235m[38;2;0;0;0m bgSkyblue [39m[49m
[38;2;106;90;205mslateblue [39m[48;2;106;90;205m[38;2;0;0;0m bgSlateblue [39m[49m
[38;2;112;128;144mslategray [39m[48;2;112;128;144m[38;2;0;0;0m bgSlategray [39m[49m
[38;2;112;128;144mslategrey [39m[48;2;112;128;144m[38;2;0;0;0m bgSlategrey [39m[49m
[38;2;255;250;250msnow [39m[48;2;255;250;250m[38;2;0;0;0m bgSnow [39m[49m
[38;2;0;255;127mspringgreen [39m[48;2;0;255;127m[38;2;0;0;0m bgSpringgreen [39m[49m
[38;2;70;130;180msteelblue [39m[48;2;70;130;180m[38;2;0;0;0m bgSteelblue [39m[49m
[38;2;210;180;140mtan [39m[48;2;210;180;140m[38;2;0;0;0m bgTan [39m[49m
[38;2;0;128;128mteal [39m[48;2;0;128;128m[38;2;0;0;0m bgTeal [39m[49m
[38;2;216;191;216mthistle [39m[48;2;216;191;216m[38;2;0;0;0m bgThistle [39m[49m
[38;2;255;99;71mtomato [39m[48;2;255;99;71m[38;2;0;0;0m bgTomato [39m[49m
[38;2;64;224;208mturquoise [39m[48;2;64;224;208m[38;2;0;0;0m bgTurquoise [39m[49m
[38;2;238;130;238mviolet [39m[48;2;238;130;238m[38;2;0;0;0m bgViolet [39m[49m
[38;2;245;222;179mwheat [39m[48;2;245;222;179m[38;2;0;0;0m bgWheat [39m[49m
[38;2;255;255;255mwhite [39m[48;2;255;255;255m[38;2;0;0;0m bgWhite [39m[49m
[38;2;245;245;245mwhitesmoke [39m[48;2;245;245;245m[38;2;0;0;0m bgWhitesmoke [39m[49m
[38;2;255;255;0myellow [39m[48;2;255;255;0m[38;2;0;0;0m bgYellow [39m[49m
[38;2;154;205;50myellowgreen [39m[48;2;154;205;50m[38;2;0;0;0m bgYellowgreen [39m[49m"
`;

exports[`use colorNames name with ansis.hex() > print all named truecolor 1`] = `
"[38;2;240;248;255maliceblue[39m
[38;2;250;235;215mantiquewhite[39m
[38;2;0;255;255maqua[39m
[38;2;127;255;212maquamarine[39m
[38;2;240;255;255mazure[39m
[38;2;245;245;220mbeige[39m
[38;2;255;228;196mbisque[39m
[38;2;0;0;0mblack[39m
[38;2;255;235;205mblanchedalmond[39m
[38;2;0;0;255mblue[39m
[38;2;138;43;226mblueviolet[39m
[38;2;165;42;42mbrown[39m
[38;2;222;184;135mburlywood[39m
[38;2;95;158;160mcadetblue[39m
[38;2;127;255;0mchartreuse[39m
[38;2;210;105;30mchocolate[39m
[38;2;255;127;80mcoral[39m
[38;2;100;149;237mcornflowerblue[39m
[38;2;255;248;220mcornsilk[39m
[38;2;220;20;60mcrimson[39m
[38;2;0;255;255mcyan[39m
[38;2;0;0;139mdarkblue[39m
[38;2;0;139;139mdarkcyan[39m
[38;2;184;134;11mdarkgoldenrod[39m
[38;2;169;169;169mdarkgray[39m
[38;2;0;100;0mdarkgreen[39m
[38;2;169;169;169mdarkgrey[39m
[38;2;189;183;107mdarkkhaki[39m
[38;2;139;0;139mdarkmagenta[39m
[38;2;85;107;47mdarkolivegreen[39m
[38;2;255;140;0mdarkorange[39m
[38;2;153;50;204mdarkorchid[39m
[38;2;139;0;0mdarkred[39m
[38;2;233;150;122mdarksalmon[39m
[38;2;143;188;143mdarkseagreen[39m
[38;2;72;61;139mdarkslateblue[39m
[38;2;47;79;79mdarkslategray[39m
[38;2;47;79;79mdarkslategrey[39m
[38;2;0;206;209mdarkturquoise[39m
[38;2;148;0;211mdarkviolet[39m
[38;2;255;20;147mdeeppink[39m
[38;2;0;191;255mdeepskyblue[39m
[38;2;105;105;105mdimgray[39m
[38;2;105;105;105mdimgrey[39m
[38;2;30;144;255mdodgerblue[39m
[38;2;178;34;34mfirebrick[39m
[38;2;255;250;240mfloralwhite[39m
[38;2;34;139;34mforestgreen[39m
[38;2;255;0;255mfuchsia[39m
[38;2;220;220;220mgainsboro[39m
[38;2;248;248;255mghostwhite[39m
[38;2;218;165;32mgoldenrod[39m
[38;2;255;215;0mgold[39m
[38;2;128;128;128mgray[39m
[38;2;0;128;0mgreen[39m
[38;2;173;255;47mgreenyellow[39m
[38;2;128;128;128mgrey[39m
[38;2;240;255;240mhoneydew[39m
[38;2;255;105;180mhotpink[39m
[38;2;205;92;92mindianred[39m
[38;2;75;0;130mindigo[39m
[38;2;255;255;240mivory[39m
[38;2;240;230;140mkhaki[39m
[38;2;255;240;245mlavenderblush[39m
[38;2;230;230;250mlavender[39m
[38;2;124;252;0mlawngreen[39m
[38;2;255;250;205mlemonchiffon[39m
[38;2;173;216;230mlightblue[39m
[38;2;240;128;128mlightcoral[39m
[38;2;224;255;255mlightcyan[39m
[38;2;250;250;210mlightgoldenrodyellow[39m
[38;2;211;211;211mlightgray[39m
[38;2;144;238;144mlightgreen[39m
[38;2;211;211;211mlightgrey[39m
[38;2;255;182;193mlightpink[39m
[38;2;255;160;122mlightsalmon[39m
[38;2;32;178;170mlightseagreen[39m
[38;2;135;206;250mlightskyblue[39m
[38;2;119;136;153mlightslategray[39m
[38;2;119;136;153mlightslategrey[39m
[38;2;176;196;222mlightsteelblue[39m
[38;2;255;255;224mlightyellow[39m
[38;2;0;255;0mlime[39m
[38;2;50;205;50mlimegreen[39m
[38;2;250;240;230mlinen[39m
[38;2;255;0;255mmagenta[39m
[38;2;128;0;0mmaroon[39m
[38;2;102;205;170mmediumaquamarine[39m
[38;2;0;0;205mmediumblue[39m
[38;2;186;85;211mmediumorchid[39m
[38;2;147;112;219mmediumpurple[39m
[38;2;60;179;113mmediumseagreen[39m
[38;2;123;104;238mmediumslateblue[39m
[38;2;0;250;154mmediumspringgreen[39m
[38;2;72;209;204mmediumturquoise[39m
[38;2;199;21;133mmediumvioletred[39m
[38;2;25;25;112mmidnightblue[39m
[38;2;245;255;250mmintcream[39m
[38;2;255;228;225mmistyrose[39m
[38;2;255;228;181mmoccasin[39m
[38;2;255;222;173mnavajowhite[39m
[38;2;0;0;128mnavy[39m
[38;2;253;245;230moldlace[39m
[38;2;128;128;0molive[39m
[38;2;107;142;35molivedrab[39m
[38;2;255;165;0morange[39m
[38;2;255;69;0morangered[39m
[38;2;218;112;214morchid[39m
[38;2;238;232;170mpalegoldenrod[39m
[38;2;152;251;152mpalegreen[39m
[38;2;175;238;238mpaleturquoise[39m
[38;2;219;112;147mpalevioletred[39m
[38;2;255;239;213mpapayawhip[39m
[38;2;255;218;185mpeachpuff[39m
[38;2;205;133;63mperu[39m
[38;2;255;192;203mpink[39m
[38;2;221;160;221mplum[39m
[38;2;176;224;230mpowderblue[39m
[38;2;128;0;128mpurple[39m
[38;2;102;51;153mrebeccapurple[39m
[38;2;255;0;0mred[39m
[38;2;188;143;143mrosybrown[39m
[38;2;65;105;225mroyalblue[39m
[38;2;139;69;19msaddlebrown[39m
[38;2;250;128;114msalmon[39m
[38;2;244;164;96msandybrown[39m
[38;2;46;139;87mseagreen[39m
[38;2;255;245;238mseashell[39m
[38;2;160;82;45msienna[39m
[38;2;192;192;192msilver[39m
[38;2;135;206;235mskyblue[39m
[38;2;106;90;205mslateblue[39m
[38;2;112;128;144mslategray[39m
[38;2;112;128;144mslategrey[39m
[38;2;255;250;250msnow[39m
[38;2;0;255;127mspringgreen[39m
[38;2;70;130;180msteelblue[39m
[38;2;210;180;140mtan[39m
[38;2;0;128;128mteal[39m
[38;2;216;191;216mthistle[39m
[38;2;255;99;71mtomato[39m
[38;2;64;224;208mturquoise[39m
[38;2;238;130;238mviolet[39m
[38;2;245;222;179mwheat[39m
[38;2;255;255;255mwhite[39m
[38;2;245;245;245mwhitesmoke[39m
[38;2;255;255;0myellow[39m
[38;2;154;205;50myellowgreen[39m"
`;
